nisystemlink/clients/testmonitor/_test_monitor_client.py

```diff
@@ -23,6 +37,159 @@ def api_info(self) -> models.ApiInfo:
+        """
+        ...
+
+    @post("query-products")
+    def query_products(
+        self, query: models.QueryProductsRequest
+    ) -> models.PagedProducts:
+        """Queries for products that match the filter.
+
+        Args:
+            query : The query contains a DynamicLINQ query string in addition to other details
+                about how to filter and return the list of products.
+
+        Returns:
+            A paged list of products with a continuation token to get the next page.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` Service or provided invalid
+                arguments.
+        """
+        ...
+
+    @returns.json  # type: ignore
+    @post("query-product-values")
+    def query_product_values(
+        self, query: models.QueryProductValuesRequest
+    ) -> List[str]:
+        """Queries for products that match the query and returns a list of the requested field.
+
+        Args:
+            query : The query for the fields you want.
+
+        Returns:
+            A list of the values of the field you requested.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` Service or provided
+            invalid arguments.
+        """
+        ...
+
+    @post("update-products", args=[Field("products"), Field("replace")])
+    def update_products(
+        self, products: List[Product], replace: bool = False
+    ) -> models.CreateProductsPartialSuccess:
+        """Updates a list of products with optional field replacement.
+
+        Args:
+            `products`: A list of products to update. Products are matched for update by id.
+            `replace`: Replace the existing fields instead of merging them.
+                If this is `True`, then `keywords` and `properties` for the product will be
+                    replaced by what is in the `products` provided in this request.
+                If this is `False`, then the `keywords` and `properties` in this request will
+                    merge with what is already present in the server resource.
+
+        Returns: A list of updates products, products that failed to update, and errors for
+            failures.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @delete("products/{id}")
+    def delete_product(self, id: str) -> Optional[ApiError]:
+        """Deletes a single product by id.
+
+        Args:
+            id (str): Unique ID of a product.
+
+        Raises:
+            ApiException: if unable to communicate with the ``/nitestmonitor`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @post("delete-products", args=[Field("ids")])
+    def delete_products(
+        self, ids: List[str]
+    ) -> Optional[models.DeleteProductsPartialSuccess]:
```

Comment:
This function either returns `DeleteProductsPartialSuccess` or results in an Exception, I believe there is no case where it returns `None`.

```suggestion
    ) -> models.DeleteProductsPartialSuccess:
```

Reply:
No, when all the deletes succeed, you get `None` returned and a 204.